app/furniture/journal/entry.rb

```diff
@@ -46,6 +49,15 @@ def self.renderer
       )
     end
 
+    def extract_terms
+      body.scan(/#(\w+)/)&.flatten&.each do |term|
+        next if journal.terms.where("aliases ILIKE ?", term)
+          .or(journal.terms.where(canonical_term: term)).exists?
+
+        journal.terms.find_or_create_by!(canonical_term: term)
+      end
```

Comment:
TBH this method feels like it wants to be a factory method on `Term`, something like `Term.extract(body, journal: journal)`  (unless we can get `journal.terms.extract(body)` to work; which would be even better)